File: Danila-Pechenev/Fourth term/Task5/CreditSystemCore/CreditSystem.cs
namespace CreditSystem;

public class CreditSystem : ICreditSystem
{
    private readonly DataStructures.ISet<Credit> set;

    public CreditSystem(DataStructures.ISet<Credit> set)
    {
        this.set = set;
    }

    public int Count()
    {
        return set.Count();
    }

    public void Add(long studentId, long courseId)
    {
        var credit = new Credit(studentId, courseId);
        set.Add(credit);
    }

    public void Remove(long studentId, long courseId)
    {
        var credit = new Credit(studentId, courseId);
        set.Remove(credit);
    }

    public bool Contains(long studentId, long courseId)
    {
        var credit = new Credit(studentId, courseId);
        return set.Contains(credit);
    }
}

File: VyacheslavBuchin/Term4/Block3/CreditSystem/CreditSystem/CreditSystem.fs
namespace CreditSystem

open ConcurrentSet
open CreditSystem.Credit

type CreditSystem(set: ISet<Credit>) =
    let credit s c = { student = s; course = c }

    interface IExamSystem with
        member this.Add student course = credit student course |> set.Add |> ignore
        member this.Contains student course = credit student course |> set.Contains
        member this.Count = set.Count()
        member this.Remove student course = credit student course |> set.Remove |> ignore
